core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/MetadataSetupService.java

```diff
@@ -361,6 +367,29 @@ public void saveLoadBalancerMetadata(Stack stack, Iterable<CloudLoadBalancerMeta
                 loadBalancerEntry.setIp(cloudLoadBalancerMetadata.getIp());
                 loadBalancerEntry.setType(cloudLoadBalancerMetadata.getType());
                 String endpoint = loadBalancerConfigService.generateLoadBalancerEndpoint(stack);
+
+                List<StackStatusView> stoppedDatalakes = stackService.getByEnvironmentCrnAndStackType(stack.getEnvironmentCrn(), StackType.DATALAKE)
+                        .stream().filter(s -> s.getStatus().getStatus().isStopState()).collect(Collectors.toList());
+                if (!stoppedDatalakes.isEmpty()) {
+                    /* Starts to check for a situation where we are resizing a datalake that did not previously have loadbalancers
+                        so that we can use the same endpoint name for a seamless transistion
+                     */
+                    LOGGER.info("Using old datalake endpoint name for resized datalake: {}, env: {}", stack.getName(), stack.getEnvironmentCrn());
+                    if (stoppedDatalakes.size() > 1) {
+                        String ids = stoppedDatalakes.stream().map(StackStatusView::getId).map(Object::toString).collect(Collectors.joining(","));
+                        LOGGER.warn("more than one datalake found to resize from: {}", ids);
+                    }
+                    Long oldId = stoppedDatalakes.get(0).getId();
```

Comment:
Could we have some query that fetches the most recently created stopped datalake or something, instead of just grabbing the first in the list? Or do we expect we'll only find a single stopped datalake in any given environment, and checking if there's more than one is just a syntactic verification?

Reply:
Correct there isn't a way for us to have more than one stopped datalake associated with an environment. The other usage of this lookup has a stronger assumption
` List<StackStatusView> res = stackService.getByEnvironmentCrnAndStackType(datahubStack.getEnvironmentCrn(), StackType.DATALAKE);
        if (res.isEmpty()) {
            return Optional.empty();
        } else {
            // it has the assumption that environment and datalake has 1:1 connection
            return Optional.of(stackService.getByIdWithListsInTransaction(res.get(FIRST).getId()));
        }`